### PR TITLE
CU-860q4n8y4 -add charts

### PR DESCRIPTION
**Pull request by Noy-Simon**
public: Add new metrics agent ( not active by default )

### CODE REVIEW

File: charts/k8s-watcher/templates/configmap-metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    [agent]
+      interval = "15s"
+      flush_interval = "1m"
+
+    [[outputs.timestream]]
+      region = "us-east-1"
+      access_key = "${TimestreamAccessKey}"
+      secret_key = "${TimestreamSecretKey}"
+
+      database_name = "prod-metrics-db"
```
Comment (erez-rabih) [inline charts/k8s-watcher/templates/configmap-metrics.yaml:23]: there are many configuration details hardcoded here which might lock us in the future if we'd like to change them
I'm referring to database name, region, retention periods, scrape interval and flush interval etc

IMO we should be receiving these from the collector endpoint for metrics configuration the same way we do with the api keys

not a lot of work but will allow us to be more agile if we need to

File: charts/k8s-watcher/templates/configmap-metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      fieldpass = ["memory_usage_bytes"," "]
+      namedrop = ["kubernetes_system_container"]
+
+    [[inputs.kube_inventory]]
+      ## URL for the Kubernetes API
+      url = "https://kubernetes.default.svc.cluster.local"
+      bearer_token = "/run/secrets/kubernetes.io/serviceaccount/token"
+      insecure_skip_verify = true
+      resource_include = [ "pods", "nodes" ]
+      fieldpass = ["resource_requests_cpu_units","resource_limits_cpu_units","resource_requests_memory_bytes","resource_limits_memory_bytes"]
```
Comment (erez-rabih) [inline charts/k8s-watcher/templates/configmap-metrics.yaml:54]: does it mean we also pass the limits and requests?
we already have this information in our DB so we might be able to drop it

File: charts/k8s-watcher/templates/daemonset-metrics.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+{{- if .Values.metrics.enabled }}
+apiVersion: apps/v1
+kind: DaemonSet
```
Comment (erez-rabih) [inline charts/k8s-watcher/templates/daemonset-metrics.yaml:3]: do we really have to install it as a daemonset on clients? won't a deployment suffice for this matter?

File: charts/k8s-watcher/templates/configmap-metrics.yaml (hunk 1 above)
Comment (Noy-Simon) [reply charts/k8s-watcher/templates/configmap-metrics.yaml:23]: will do

File: charts/k8s-watcher/templates/configmap-metrics.yaml (hunk 2 above)
Comment (Noy-Simon) [reply charts/k8s-watcher/templates/configmap-metrics.yaml:54]: we talked about it and decided to collect it for now
we can change it in the future if we want

File: charts/k8s-watcher/templates/daemonset-metrics.yaml (hunk 3 above)
Comment (Noy-Simon) [reply charts/k8s-watcher/templates/daemonset-metrics.yaml:3]: we wanted to support large costumers 
so we decided to use Daemonset

File: charts/k8s-watcher/templates/configmap-metrics.yaml (hunk 1 above)
Comment (Noy-Simon) [reply charts/k8s-watcher/templates/configmap-metrics.yaml]: done

**Review by erez-rabih — APPROVED** [review]

**Pull request merged**